### ISSUE TITLE
Additional coding lessons later in the semester

### ISSUE THREAD

**Issue by ajrbrennan** [paultopia/gobbledygook#4]
One thing that I struggled with throughout the semester was feeling like I knew _what_ I wanted to do, but not knowing _how_ to do it. I struggled with manipulating csv files -- adding columns, changing names of columns, changing information in a column to plain language (instead of 1 and 0). A better grasp of those skills would have let me focus more on the statistical analysis side of things. 

I understand why we started with coding, and as difficult as the first problem set was, I appreciate that it forced me to get comfortable with Python quickly. However, I think that some of the things I learned early on (for loops, for example), I didn't use later. Maybe my issues with csvs would have been alleviated if the coding portions of the class had focused from the beginning on working with csv files. 

The datacamp material was helpful, but I always had a better grasp of the specific code I needed to be using after working through examples in class. More emphasis on the coding side throughout the semester I think would make a big difference.

**Comment by emdavidson** [comment]
Yes totally! Like we could write the pseudo-code but when it came to some of the trickier stuff like manipulating the datasets, it would have been great to have more back-and-forth regarding how to make these things happen. I was thinking that Prof could hold some time for each week to go over code, or we could submit questions about syntax through a discussion forum.

**Comment by jarred-stindt** [comment]
I agree with the coding issue, in part because coding is what I struggled with the most in the class, but mainly because it seems like we spent 3 weeks of in depth coding, and have used almost none of that outside of the first assignment.  My suggestion, instead of adding more coding later, would be to lighten up the coding at the beginning, focus on the areas that are necessary to sort through data, and focus way less on areas that we will literally never use again I think that would also help the course load in general, and we could get into the data analysis side sooner.

**Comment by samkbaier** [comment]
Maybe the addition of small, ungraded practice problems (for students to practice coding) would help address the "I know what I want to do but it's taking me forever to figure out how to do it" issue.  Students could be encouraged to save and review the bits of code they write to solve those practice problems, with the hope that much of it could be modified by the student for later use on graded problem sets.  Having a few more ideas and having to build less from scratch while working on a problem set would be a big help.
